src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -281,6 +278,11 @@ private bool HasNameInMetadata
             out bool isBad)
         {
             var typeWithModifiers = TypeSymbolWithAnnotations.Create(type, CSharpCustomModifier.Convert(customModifiers));
+            // PROTOTYPE(NullableReferenceTypes): Avoid setting IsNullable in TypeSymbolWithAnnotations.Create
+            // only to undo that in SetUnknownNullabilityForReferenceTypesIfNecessary. Same comment applies
+            // to other uses of SetUnknownNullabilityForReferenceTypesIfNecessary.
+            typeWithModifiers = typeWithModifiers.SetUnknownNullabilityForReferenceTypesIfNecessary(moduleSymbol);
```

Comment:
>typeWithModifiers = typeWithModifiers.SetUnknownNullabilityForReferenceTypesIfNecessary(moduleSymbol); [](start = 12, length = 102)

This looks like a wrong place to do this, constructors should handle this.

Reply:
The `PEParameterSymbol` constructors? Why there?

---
In reply to: [171136573](https://github.com/dotnet/roslyn/pull/25087#discussion_r171136573) [](ancestors = 171136573)

Reply:
Because you already do this there, but only on one branch, both branches should do this, the other branch is for the case of no attribute. 

---
In reply to: [171150806](https://github.com/dotnet/roslyn/pull/25087#discussion_r171150806) [](ancestors = 171150806,171136573)